src/main/java/com/rapid7/client/dcerpc/mslsad/LocalSecurityAuthorityService.java

```diff
@@ -95,5 +99,24 @@ public LsarLookupNamesResponse lookupNames(final ContextHandle policyHandle, fin
         final LsarLookupNamesRequest request = new LsarLookupNamesRequest(policyHandle, names);
         return callExpectSuccess(request, "LsarLookupNames");
     }
+
+    public String[] lookupSIDs(ContextHandle policyHandle, String... SIDs)
+        throws IOException, MalformedSIDException
+    {
+        String[] mappedNames;
+        final LsarLookupSIDsRequest request =
+            new LsarLookupSIDsRequest(policyHandle, SIDs);
+        final LsarLookupSIDsResponse lsarLookupSIDsResponse = callExpect(request, "LsarLookupSIDs",
+                SystemErrorCode.ERROR_SUCCESS,
+                SystemErrorCode.STATUS_SOME_NOT_MAPPED);
+
+        LSAPRTranslatedName[] nameArray = lsarLookupSIDsResponse.getLsaprTranslatedNames().getlsaprTranslatedNameArray();
+        mappedNames = new String[nameArray.length];
```

Comment:
you don't need to declare the field at the start. just do it here
```
final String[] mappedNames = new String[nameArray.length];
```

Reply:
Done. I was initially declaring it and then using mappedCount to set the size. Changed it when we decided to pass around nulls when a mapping is not found.